### DISCUSSION TITLE
Sonarqube not reporting results back to Jenkins and Gerrit

### DISCUSSION THREAD

**Discussion by navyashetty** [jenkinsci/sonar-gerrit-plugin#149] [Q&A] [answered]
### Jenkins and plugins versions report

I have a Jenkins freestyle job that is triggered as downstream job. When it is triggered, it downloads the changes and runs a Sonarqube Analysis on them. After the analysis is done, it is supposed to post comments on Gerrit about the results.

Unfortunately, no matter what the result of the Sonarqube analysis, the comment always says "sonarqube violations have not been found."

Sonar analysis report
![image](https://github.com/jenkinsci/sonar-gerrit-plugin/assets/8750680/c380102a-d8b6-4829-89b1-4b8bc2dee6a5)

Jenkins console output
![image](https://github.com/jenkinsci/sonar-gerrit-plugin/assets/8750680/7e510ca3-bd4f-49d1-9e8c-7a78d8bab953)

Job configuration
![image](https://github.com/jenkinsci/sonar-gerrit-plugin/assets/8750680/dd735dea-d3b0-4ff5-97a9-59ca2ba58dab)

Sonarqube version: 9.9
Jenkins version: 2.414.1
sonargerrit plugin version: 383.v1e1ea_b_da_460d
Gerrit Trigger plugin version: 2.39.1
SonarQube Scanner for Jenkins version: 2.15

Could you please suggest what is the issue here ?

### What Operating System are you using (both controller, and any agents involved in the problem)?

ubuntu

### Reproduction steps

Enabled webhook in sonarqube but still it is not working.

### Expected Results

![image](https://github.com/jenkinsci/sonar-gerrit-plugin/assets/8750680/82f4f8c7-f83f-45b1-bc03-13382fd94b96)

### Actual Results

![image](https://github.com/jenkinsci/sonar-gerrit-plugin/assets/8750680/1a401ece-a96c-4d51-b2b5-cb2fe1d52f43)

**Comment by climblinne** [comment]
I got the same issue. Looking at the <sonarqube_server>/api/issues/search?components=<PROJECT_KEY>&pullRequest=<PULL_REQUEST> I see the list of issues. But the "Issues to be commented" is zero. 
Do you found a solution?

**Comment by SR4ven** [comment]
I have the exact same problem.

Sonarqube version: 10.2.1
Jenkins version: 2.414.3
sonargerrit plugin version: 385.v890738eeff15
Gerrit Trigger plugin version: 2.39.3
SonarQube Scanner for Jenkins version: 2.16.1

**Comment by navyashetty** [comment, marked as answer]
In my case, the problem was resolved by providing the necessary access to the SonarQube project for the service user used during the SonarQube analysis. This ensured that the analysis job could access and retrieve the relevant data.

**Comment by reda-alaoui** [comment]
Closing based on https://github.com/jenkinsci/sonar-gerrit-plugin/issues/147#issuecomment-1790615221

**Comment by SR4ven** [comment]
The problem was indeed the missing "Browse" permission for the service user.

Might be worth mentioning that requirement in the readme.
